### PR TITLE
ce_3

### LINKED ISSUE #20: ce_3
Description: Data subsetting

Script URL: https://www.ncl.ucar.edu/Applications/Scripts/ce_3.ncl

Plot URL: https://www.ncl.ucar.edu/Applications/cylineq.shtml

### PR DESCRIPTION
**Pull request by jukent**
Closes #20 

ce_3_1  
NCL: https://www.ncl.ucar.edu/Applications/Images/ce_3_1_lg.png

image

and ce_3_2
NCL: https://www.ncl.ucar.edu/Applications/Images/ce_3_2_lg.png
image

### CODE REVIEW

**Comment by clyne** [comment]
@erogluorhan just a friendly reminder that this is ready for your review.

**Comment by erogluorhan** [comment]
Comparison of Python Plots to NCL Originals:

1) Python-generated plots look pretty similar to NCL originals. However, ranging of the colors in colormaps do not resemble NCL colormaps (independent from what colors are used). For example, there is a very narrow Cyan band around 13.13 in Python colormaps that lead to a narrow part in Python plots (near the bottom). However there is no such narrow range of any color around 13 in NCL colormaps, and no narrow part is on NCL plots as well. These sort of differences make it hard to say replicated plots are 100% equal to NCL ones. Is it possible to generate a colormap of same or closer color ranges (whatever the used colors are)?

2) It would be better to provide same colorbar values with NCL colormaps (e.g. show 30 instead of 29.54 and so on).

3) Provide the main plot title for Python plots, which is apparent in NCL.

**Comment by mnlevy1981** [comment]
For the colormap, you could do `import cmaps` and then use `cmaps.BlAqGrYeOrRe` in your plots. And for the tickmarks, you can add `'ticks' : range(0, 31, 2)` to `cbar_kwargs`.

File: Plots/Contours/NCL_ce_3_2_lg.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"""
```
Comment (erogluorhan) [inline Plots/Contours/NCL_ce_3_2_lg.py]: Review comments for above function `Plots/Contours/NCL_ce_3_1_lg.py` can be similarly applied to this function as well.

**Comment by jukent** [comment]
Screen Shot 2020-02-11 at 5 26 06 PM

There are 2 NCL plots: one demonstrating default tick markers and the other how to manipulate them. I had initially manipulated both of my plots to match the NCL plots, but I think it is better to comceptually match the description by having one plot where I do not manipulate ticks (ce_3_1) and the other where I do (ce_3_2). Because of this ce_3_1 looks less like the NCL example than before. Both plots now have suptitles and sensible integer colorbar labels.

File: Plots/Contours/NCL_ce_3_2_lg.py (hunk 1 above)
Comment (erogluorhan) [reply Plots/Contours/NCL_ce_3_2_lg.py]: Addressed them with commit [https://github.com/jukent/GeoCAT-examples/commit/33aaeca2c09e8763983cb8ca1b7a4b26c4459a1c](url)

File: Plots/Contours/NCL_ce_3_1_lg.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+"""
+proj_1_lg
+===============
+Plots/Contours/Lines
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply Plots/Contours/NCL_ce_3_1_lg.py]: Addressed them now with a commit

File: Plots/Contours/NCL_ce_3_1_lg.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+xlabels = ['30E', '60E', '90E', '120E']
+yticks = [-60, -30, 0, 30]
+ylabels = ['60S', '30S', '0', '30N']
+plt.xticks(xticks, xlabels)
+plt.yticks(yticks, ylabels)
+plt.tick_params(which='both',right=True, top=True)
+plt.minorticks_on()
+
+# set titles and axis labels
+plt.title('Potential Temperature                  Celsius', fontsize=15)
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply Plots/Contours/NCL_ce_3_1_lg.py]: Addressed them now with a commit

File: Plots/Contours/NCL_ce_3_1_lg.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                    levels = 40, vmin=0, vmax=32, cmap = 'gist_rainbow_r', 
+                    cbar_kwargs={"label":'', "shrink":0.8})
+
+# set axis limits
+ax.set_xlim([30,120])
+ax.set_ylim([-60,30])
+
+# manually specify ticks
+xticks = [30, 60, 90, 120]
+xlabels = ['30E', '60E', '90E', '120E']
```
[the remark this answers is not in the document]
Comment (erogluorhan) [reply Plots/Contours/NCL_ce_3_1_lg.py]: Addressed them now with a commit

File: Plots/Contours/NCL_ce_3_2_lg.py (hunk 1 above)
Comment (erogluorhan) [reply Plots/Contours/NCL_ce_3_2_lg.py]: NCL_ce_3_1_lg:

image

NCL_ce_3_2_lg:

image

**Pull request merged**